simple-vm/terraform/main.tf

```diff
@@ -0,0 +1,86 @@
+variable "project_id" {
+  description = "GCP project ID"
+}
+
+variable "region" {
+  description = "GCP region"
+}
+
+variable "zone" {
+  description = "GCP zone"
+}
+
+variable "gce_ssh_user" {
+  description = "User to create on the GCE instance"
+}
+
+variable "ssh_private_key_file" {
+  description = "Path to SSH private key"
+}
+
+variable "ssh_public_key_file" {
+  description = "Path to SSH public key"
+}
+
+provider "google" {
+  project = var.project_id
+  region  = var.region
+  zone    = var.zone
+}
+
+provider "cloudflare" {
+}
```

Comment:
The provider has no variables defined in it. Do we still need to define the provider?

Reply:
Yes, we require the empty block so that Terraform realizes it needs to install the Cloudflare provider plugin when doing `terraform init`.

I couldn't find an exact line in docs, but found this in the official blog
> Note that you need to leave the empty provider definition in the file, so that Terraform knows to install the Cloudflare plugin.

Source: https://blog.cloudflare.com/getting-started-with-terraform-and-cloudflare-part-1/

Reply:
Huh, apparently the information I had and the blog post seems to be outdated. I tried it out, by removing the empty block, it worked! 🎉 

Also found this in the docs - 
> Unlike many other objects in the Terraform language, a provider block may be omitted if its contents would otherwise be empty. Terraform assumes an empty default configuration for any provider that is not explicitly configured.

Source: https://www.terraform.io/docs/language/providers/configuration.html